scripts/ui/inventory/inventory.gd

```diff
@@ -0,0 +1,141 @@
+	equip_slots[0].slot_type = SlotClass.SlotType.HELMET
+	equip_slots[1].slot_type = SlotClass.SlotType.BODY
+	equip_slots[2].slot_type = SlotClass.SlotType.BOOTS
+	equip_slots[3].slot_type = SlotClass.SlotType.WEAPONS
+	equip_slots[4].slot_type = SlotClass.SlotType.WEAPONS
+	initialize_inventory()
+	initialize_equip()
+	update_active_item_label()
+		
+func update_active_item_label():
+	var slots = inventory_slots.get_children()
+	tooltip.visible = true
+	#if PlayerInventory.active_item_slot == null or slots[PlayerInventory.active_item_slot].slot_type == SlotClass.SlotType.INVENTORY:
+		#return
+	if slots[PlayerInventory.active_item_slot].item != null:
+		tooltip.visible = true
+		var info_slots = get_tree().get_nodes_in_group("Info_Slots")
+		info_slots[0].text = JsonData.item_data[slots[PlayerInventory.active_item_slot].item.item_name]["ItemCategory"]
+		info_slots[1].text = slots[PlayerInventory.active_item_slot].item.item_name
+		info_slots[2].text = JsonData.item_data[slots[PlayerInventory.active_item_slot].item.item_name]["Description"]
+		if JsonData.item_data[slots[PlayerInventory.active_item_slot].item.item_name]["ItemCategory"] == "Consumable":
+			info_slots[3].text = "Adds " + str(JsonData.item_data[slots[PlayerInventory.active_item_slot].item.item_name]["AddHealth"]) + " Health"
+			info_slots[4].text = "Adds " + str(JsonData.item_data[slots[PlayerInventory.active_item_slot].item.item_name]["AddEnergy"]) + " Energy"
+		elif JsonData.item_data[slots[PlayerInventory.active_item_slot].item.item_name]["ItemCategory"] == "Tool":
+			info_slots[3].text = "Adds " + str(JsonData.item_data[slots[PlayerInventory.active_item_slot].item.item_name]["AddRepair"]) + " Repair"
+			info_slots[4].text = "Adds " + str(JsonData.item_data[slots[PlayerInventory.active_item_slot].item.item_name]["AddEnergy"]) + " Energy"
+		elif JsonData.item_data[slots[PlayerInventory.active_item_slot].item.item_name]["ItemCategory"] == "Weapon":
+			info_slots[3].text = "Damage: " + str(JsonData.item_data[slots[PlayerInventory.active_item_slot].item.item_name]["ItemAttack"])
+			info_slots[4].text = "Reload: " + str(JsonData.item_data[slots[PlayerInventory.active_item_slot].item.item_name]["ItemReload"]) + "s"
+		else:
+			info_slots[3].text = ""
+			info_slots[4].text = ""
+	else:
+		tooltip.visible = false
+		
+func initialize_inventory():
+	var slots = inventory_slots.get_children()
+	for i in range(slots.size()):
+		if PlayerInventory.inventory.has(i):
+			slots[i].initialize_item(PlayerInventory.inventory[i][0], PlayerInventory.inventory[i][1])
+
+func initialize_equip():
+	var es = equip_slots
+	for i in range(es.size()):
+		if PlayerInventory.equips.has(i):
+			es[i].initialize_item(PlayerInventory.equips[i][0], PlayerInventory.equips[i][1])
+
+func slot_gui_input(event: InputEvent, slot: SlotClass):
+	if event is InputEventMouseButton:
+		if event.button_index == MOUSE_BUTTON_LEFT && event.pressed:
+			if find_parent("UI").holding_item != null:
```

Comment:
Instead of calling find_parent a billion times, call it once and store it in a variable.